hooks/useWeatherData.ts

```diff
@@ -74,11 +58,11 @@ export const useWeatherData = (city: string, today: boolean) => {
     }
 
     const temps = forecastData.list
-      .filter((item) => {
+      .filter((item: ForecastItem) => {
```

Comment:
Ej dobra, teraz widzę, że Ty w ogóle nie używasz tych scheme z zoda. Generalnie to, że dostajemy z nich typy to jest bonus. Najważniejsze jest to, że one dają nam sposób na sprawdzenie czy obiekt, który dostajemy gdzieś z zewnątrz (na przykład z API) faktyczni ma taka strukturę, jakiej oczekujemy. No i jak już to sprawdzimy, korzystając z np. [parse](https://github.com/colinhacks/zod?tab=readme-ov-file#parse), wtedy możemy sobie uzywać tego typu z `z.infer`. Ale najczęściej w ogóle nie będziemy musieli, bo to będzie już dalej inferowane przez TSa